translate.c

```diff
@@ -201,3 +205,27 @@ int translate_open(const char *path, int flags)
 	errno = ENOENT;
 	return -1;
 }
+
+/* linux-firmware uses .zst as file extension */
+#define ZSTD_EXTENSION ".zst"
+
+/**
+ * open_maybe_compressed() - open a file and maybe decompress it if necessary
+ * @filename:	path to a file that may be compressed (should not include compression format extension)
+ *
+ * Return: opened fd on success, -1 on error
+ */
+static int open_maybe_compressed(const char *path)
+{
+	int fd = -1;
+
+	if (access(path, F_OK) == 0) {
+		fd = open(path, O_RDONLY);
+	} else {
+		char path_with_zstd_extension[PATH_MAX] = {0};
+		snprintf(path_with_zstd_extension, PATH_MAX, "%s%s", path, ZSTD_EXTENSION);
```

Comment:
`asnprintf` is easier

Reply:
I was considering using `asprintf` initially but went with `snprintf` as I assume the former internally calls malloc and uses the heap, which I would think means more overhead and in this case we already have a good max size for the string readily available so we can use the stack instead. Correct me if I'm wrong though, I'm not that familiar with C.

Either way, I can't find any manpage for `asnprintf`, but I do find it mentioned elsewhere. It seems like it comes from GNU libunistring? Is this what you meant, or did you mean `asprintf`?

Reply:
`asprintf`, of course. 'n' there doesn't make sense.